### PR TITLE
[Workspace] Delete the virtual global workspace

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

Delete the virtual global workspace.
When workspace feature flag is enabled and user has not created any workspace, only OSD admin user can see any legacy saved object (not belonging to any workspace) on the home page. Non OSD admin users should not see the legacy saved objects if they can not access any workspace.
### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7095

## Testing the changes

1. Clone the latest osd code and run yarn osd bootstrap
2. Modify config/opensearch_dashboards.yml
```yml
workspace.enabled: true 
savedObjects.permission.enabled: true
```
3. Run yarn start --no-base-path
## Changelog
- feat: [Workspace] Delete the virtual global workspace

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts [hunk 1]
```diff
@@ -513,15 +514,13 @@ export class WorkspaceSavedObjectsClientWrapper {
...
         } else {
-          /**
-           * If no workspaces present, find all the docs that
-           * ACL matches read / write / user passed permission
-           */
-          options.ACLSearchParams.permissionModes = getDefaultValuesForEmpty(
-            options.ACLSearchParams.permissionModes,
-            [WorkspacePermissionMode.Read, WorkspacePermissionMode.Write]
-          );
-          options.ACLSearchParams.principals = principals;
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts:524]: The code here is used for searching user object and could we keep it as it is? We can move the return block to line 460 maybe?

Comment (yubonluo) [reply src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts]: Updated

**Review by ruanyl — APPROVED** [review]

File: src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ import {
   SavedObjectsServiceStart,
   SavedObjectsClientContract,
   SavedObjectsDeleteByWorkspaceOptions,
+  SavedObjectsFindResponse,
```
Comment (ruanyl) [inline src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts]: Seems this is not used

**Review by ruanyl — APPROVED** [review]

**Comment by ruanyl** [comment]
I like these changes, after removing the special "artificial" workspace, the code smells much better 😃

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**